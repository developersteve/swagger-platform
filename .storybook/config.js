import { configure } from '@storybook/react';
import { createStories } from './createStories';
import { withKnobs } from '@storybook/addon-knobs';
import { checkA11y } from '@storybook/addon-a11y';
import { withViewport } from '@storybook/addon-viewport';
import { setOptions } from '@storybook/addon-options';
import { withBackgrounds } from '@storybook/addon-backgrounds';
import { addDecorator } from '@storybook/react';
addDecorator(withKnobs);
addDecorator(checkA11y);
addDecorator(withViewport('responsive'));
addDecorator(withBackgrounds());
setOptions({
  name: 'OpenAPI Platform',
});
configure(() => {
  const tsxModuleContext = require.context('../src/view', true, /\.tsx$/);
  const moduleInfo = tsxModuleContext
    .keys()
    .map(path => ({ path, requiredModule: tsxModuleContext(path) }));
  createStories(moduleInfo);
}, module);
