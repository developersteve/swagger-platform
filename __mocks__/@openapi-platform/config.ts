const { schema, apiBaseUrl, uiUrl, serverUrl } = require.requireActual(
  '@openapi-platform/config',
);
export { schema, apiBaseUrl, uiUrl, serverUrl };
/**
 * Just provides a consistent config for tests to use
 */
export function readConfig() {
  const config = schema.load({
    env: 'test',
    server: {
      port: 8080,
      useCors: false,
    },
    database: {
      name: 'test_name',
      host: 'test_host',
      port: 1337,
      username: 'test_username',
      password: 'test_password',
    },
  });
  return config;
}
