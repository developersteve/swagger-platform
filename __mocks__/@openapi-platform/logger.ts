import { mockFunctions } from 'jest-mock-functions';
const actualModule = require.requireActual('@openapi-platform/logger');
export const {
  openapiLogger,
  overrideConsoleLogger,
  overrideUtilInspectStyle,
} = mockFunctions(actualModule);
openapiLogger.mockImplementation((...options) => {
  return mockFunctions(actualModule.openapiLogger(...options), { recursive: true });
});
