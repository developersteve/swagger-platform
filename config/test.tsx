export const testConfig = {
  frontend: {
    baseApiUrl: 'http://localhost:8111/',
  },
  backend: {
    port: 8112,
    useCors: true,
    initDummyData: true,
    databaseName: '',
    databaseHost: '',
    databasePort: 5432,
    databaseUsername: '',
    databasePassword: '',
  },
};
