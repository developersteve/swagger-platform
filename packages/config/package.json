{
  "name": "@openapi-platform/config",
  "description": "Config parser, let's you read openapi-platform.config.zzz config files",
  "version": "1.0.0-alpha.0",
  "main": "./lib/index.js",
  "types": "./src/index.ts",
  "license": "Apache-2",
  "private": false,
  "sideEffects": false,
  "dependencies": {
    "js-yaml": "^3.12.0",
    "json5": "^2.0.1",
    "convict": "^4.3.2",
    "fs-extra": "^7.0.0"
  }
}
