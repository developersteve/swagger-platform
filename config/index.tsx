import { developmentConfig } from 'config/development';
import { productionConfig } from 'config/production';

interface Configuration {
  /**
   * Frontend configuration options.
   */
  readonly frontend: {
    /**
     * The base API URL used for all requests.
     */
    readonly baseApiUrl: string;
  };

  /**
   * Backend configuration options.
   */
  readonly backend: {
    /**
     * The port number used for incoming connections.
     */
    readonly port: number;

    /**
     * Whether or not CORS requests should be allowed.
     */
    readonly useCors: boolean;
  };
}

let configTemp: Configuration | undefined;
if (process.env.NODE_ENV === 'development') {
  configTemp = developmentConfig;
} else if (process.env.NODE_ENV === 'production') {
  configTemp = productionConfig;
} else {
  throw new Error('Unknown environment type');
}

export const config = configTemp as Configuration;
