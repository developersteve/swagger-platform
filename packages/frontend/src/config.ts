import { readConfig } from '@openapi-platform/config';
export const config = readConfig();
