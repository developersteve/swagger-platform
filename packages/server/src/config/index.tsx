import { developmentConfig } from './development';
import { productionConfig } from './production';
import { testConfig } from './test';

interface Configuration {
  /**
   * Frontend configuration options.
   */
  readonly frontend: {
    /**
     * The base API URL used for all requests.
     */
    readonly baseApiUrl: string;
  };

  /**
   * Backend configuration options.
   */
  readonly backend: {
    /**
     * The port number used for incoming connections.
     */
    readonly port: number;

    /**
     * Whether or not CORS requests should be allowed.
     */
    readonly useCors: boolean;

    /**
     * Whether or not dummy data should be created for development purposes.
     */
    readonly initDummyData: boolean;

    /**
     * The name of the PostgreSQL database to connect to.
     */
    readonly databaseName: string;

    /**
     * The hostname of the PostgreSQL database to connect to.
     */
    readonly databaseHost: string;

    /**
     * The port of the PostgreSQL database to connect to.
     */
    readonly databasePort: number;

    /**
     * The username of the PostgreSQL database to connect to.
     */
    readonly databaseUsername: string;

    /**
     * The password of the PostgreSQL database to connect to.
     */
    readonly databasePassword: string;
  };
}

let configTemp: Configuration | undefined;
if (process.env.NODE_ENV === 'development') {
  configTemp = developmentConfig;
} else if (process.env.NODE_ENV === 'production') {
  configTemp = productionConfig;
} else if (process.env.NODE_ENV === 'test') {
  configTemp = testConfig;
} else {
  throw new Error('Unknown environment type: ' + process.env.NODE_ENV);
}

export const config = configTemp as Configuration;
