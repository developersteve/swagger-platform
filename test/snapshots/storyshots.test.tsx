import { createMount } from '@material-ui/core/test-utils';
import initStoryshots from '@storybook/addon-storyshots';
import registerRequireContextHook from 'babel-plugin-require-context-hook/register';
import { join } from 'path';
import { readConfig, apiBaseUrl } from '@openapi-platform/config';
jest.mock('@openapi-platform/config');
const config = readConfig();
global.API_BASE_URL = apiBaseUrl(config);
// Required because Storybook uses require.context but this file is not run through Webpack
registerRequireContextHook();

initStoryshots({
  configPath: join(__dirname, '../../.storybook'),
  /*
    If you remove this, you end up some rather confusing errors.
    See: https://material-ui.com/guides/testing/ and https://github.com/mui-org/material-ui/issues/9243
  */
  renderer: createMount,
});
