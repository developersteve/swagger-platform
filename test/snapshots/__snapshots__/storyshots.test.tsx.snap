// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots SdkConfigModal Submit 1`] = `[Function]`;

exports[`Storyshots SimpleToolbar Generic 1`] = `[Function]`;

exports[`Storyshots SpecModal Submit 1`] = `[Function]`;
